frontend/src/metabase/browse/utils.ts

```diff
@@ -136,3 +145,52 @@ export const filterModels = (
     unfilteredModels,
   );
 };
+
+type CollectionPrefs = Partial<Record<CollectionId, ModelVisibilityPrefs>>;
+
+type ModelVisibilityPrefs = {
+  expanded: boolean;
+  showAll: boolean;
+};
+
+const isRecordWithCollectionIdKeys = (
+  prefs: unknown,
+): prefs is Record<CollectionId, any> =>
+  typeof prefs === "object" &&
```

Comment:
We should have some better type utils. This would incorrectly work with an array.